packages/core/src/routes/interaction/consent/index.ts

```diff
@@ -225,9 +224,8 @@ export default function consentRoutes<T extends IRouterParamContext>(
       const { accountId } = session;
 
       const application =
-        clientId === demoAppApplicationId
-          ? buildDemoAppDataForTenant('')
-          : await queries.applications.findApplicationById(clientId);
+        buildBuiltInApplicationDataForTenant('', clientId) ??
```

Comment:
ditto: From the method name `buildBuiltInApplicationDataForTenant`, it should only take the built-in application's `clientId`.  Maybe a util method `isBuiltInClientId`  and a literal type `BuiltInClientIds`

Reply:
updated